packages/rpc-core/src/rpc-methods/__tests__/simulate-transaction-test.ts

```diff
@@ -0,0 +1,717 @@
+        expect.assertions(1);
+        const { value: latestBlockhash } = await rpc.getLatestBlockhash().send();
+        const message = getMockTransactionMessage({
+            blockhash: latestBlockhash.blockhash,
+            feePayerAddressBytes: MOCK_PUBLIC_KEY_BYTES,
+            memoString: `Hello from the web3.js tests! [${performance.now()}]`,
+        });
+        const signature = new Uint8Array(Array(64).fill(0));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                {
+                    commitment: 'processed',
+                    encoding: 'base64',
+                    sigVerify: false,
+                }
+            )
+            .send();
+
+        await expect(resultPromise).resolves.toMatchObject({
+            value: expect.objectContaining({
+                accounts: null,
+                err: null,
+                logs: expect.any(Array),
+                returnData: null,
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('returns a BlockhashNotFound error when the blockhash does not exist when `replaceRecentBlockhash` is false', async () => {
+        expect.assertions(1);
+        const secretKey = await getSecretKey();
+        const message = getMockTransactionMessage({
+            blockhash: base58.deserialize(new Uint8Array(Array(32).fill(0)))[0],
+            feePayerAddressBytes: MOCK_PUBLIC_KEY_BYTES,
+            memoString: `Hello from the web3.js tests! [${performance.now()}]`,
+        });
+        const signature = new Uint8Array(await crypto.subtle.sign('Ed25519', secretKey, message));
+        const resultPromise = rpc
+            .simulateTransaction(
+                Buffer.from(
+                    new Uint8Array([
+                        0x01, // Length of signatures
+                        ...signature,
+                        ...message,
+                    ])
+                ).toString('base64') as Base64EncodedWireTransaction,
+                {
+                    commitment: 'processed',
+                    encoding: 'base64',
+                    replaceRecentBlockhash: false,
+                }
+            )
+            .send();
+
+        await expect(resultPromise).resolves.toMatchObject({
+            value: expect.objectContaining({
+                accounts: null,
+                err: 'BlockhashNotFound',
+                logs: expect.any(Array),
+                returnData: null,
+                unitsConsumed: expect.any(BigInt),
+            }),
+        });
+    });
+
+    it('replaces the invalid blockhash when `replaceRecentBlockhash` is true', async () => {
```

Comment:
Just continuing on my comment of the mutually exclusive configs, if we're going to keep the ability to provide both, we should have a test that observes the API's behavior if both are provided. 
Although I think we shouldn't 😝

Reply:
That's a type error:

<img width="612" alt="Screenshot 2023-08-21 at 18 14 02" src="https://github.com/solana-labs/solana-web3.js/assets/1711350/69f016bc-2f89-4f82-95b6-07e8d63dda72">
<img width="612" alt="Screenshot 2023-08-21 at 18 14 14" src="https://github.com/solana-labs/solana-web3.js/assets/1711350/40240dbc-ce1a-4524-ab07-a8ccc30200e5">

FWIW the RPC message is:
```
    "error": {
        "code": -32602,
        "message": "sigVerify may not be used with replaceRecentBlockhash"
    },
```

But I think a type error is definitely better!

Reply:
Yeah, that's a rough error. But nonetheless it is a type error!

I'm starting to think _a lot_ of our type errors for parameters are going to be balls of yarn to decode for the unassuming developer. Perhaps later we can comb through and try to reduce suffering across many parameter types.

Regardless, making sure we got a type error instead of RPC error was my main concern, thanks for clarifying with this example!